WordPressFlux/WordPressFlux/Sources/QueryStore.swift

```diff
@@ -58,4 +113,62 @@ open class QueryStore<State, Query>: StatefulStore<State>, Unsubscribable {
     open func queriesChanged() {
         // Subclasses should implement this
     }
+
+}
+
+// In ideal world, those wouldn't be extensions on `Encodable`/`Codable`, but rather refinements of
+// `QueryStore` in which `Store` conforms to `Codable` — then we would just change behavior _inside_ the `QueryStore`,
+// based on whether the type conforms to `Codable` or not.
+// Regretfully, Swift type system (or my Swift skills) does't allow to express that —
+// you can't just do `let codableSelf = self as? QueryStore<Codable, Query>` because
+// _Codable itself doesn't conform to Codable_.
+private extension Encodable {
+    func storeCachedState(at url: URL) {
+        do {
+            let jsonEncoder = JSONEncoder.init()
+            let encodedStore = try jsonEncoder.encode(self)
+
+            try encodedStore.write(to: url, options: [.atomic])
+        } catch {
```

Comment:
I'm not sure I'd `catch` the error here or propagate it and handle it on the store. In any case, we should use `DDlogError` instead.